src/agoradatatools/reporter.py

```diff
@@ -0,0 +1,130 @@
+
+        Args:
+            **kwargs: Keyword arguments for the DatasetReport object.
+        """
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    @classmethod
+    def format_link(cls, syn_id: str, version: int) -> str:
+        """Generates a link to a specific version of a synapse entity
+
+        Args:
+            syn_id (str): Synapse ID of the entity.
+            version (int): Version number of the entity.
+
+        Returns:
+            str: Formatted link to the specific version of the entity.
+        """
+        return f"https://www.synapse.org/Synapse:{syn_id}.{version}"
+
+
+@dataclass
+class ADTGXReporter:
+    """
+    Holds all DatasetReport objects needed to update the Synapse table for an ADT run.
+    Includes attributes necessary to execute the upload to the table, as well as data which is common
+    to all DatasetReport objects.
+
+    Attributes:
+        syn: Synapse session object.
+        platform: The platform where the processing was run.
+        run_id: The id of the processing run. This will be passed from the `process` CLI command.
+        table_id: Synapse ID of the Synapse table to be updated.
+        data_manifest_file: Synapse ID of the ADT manifest file.
+        data_manifest_version: Version number of the ADT manifest file.
+        data_manifest_link: URL of the specific version of the ADT manifest file.
+        reports: List of DatasetReport objects to be added to the table.
+    """
+
+    syn: synapseclient.Synapse
+    platform: Platform
+    run_id: str
+    table_id: str
+    data_manifest_file: Optional[str] = field(default=None)
+    data_manifest_version: Optional[int] = field(default=None)
+    data_manifest_link: Optional[str] = field(default=None)
+    reports: Optional[List[DatasetReport]] = field(default_factory=list)
+
+    def add_report(self, report: DatasetReport) -> None:
+        """Adds a DatasetReport object to the list of reports.
+
+        Args:
+            report (DatasetReport): DatasetReport object to be added to the list.
+        """
+        self.reports.append(report)
+
+    def _update_reports_before_upload(self) -> None:
+        """Updates the DatasetReport objects with common attributes before uploading to the table."""
+        timestamp = datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        for report in self.reports:
+            report.set_attributes(
+                timestamp=timestamp,
+                platform=self.platform.value,
+                run_id=self.run_id,
+                data_manifest_file=self.data_manifest_file,
+                data_manifest_version=self.data_manifest_version,
+                data_manifest_link=self.data_manifest_link,
+            )
+
+    def update_table(self) -> None:
+        """Updates the Synapse table adding one new row for each DatasetReport object if the platform is not LOCAL."""
+        if self.platform != Platform.LOCAL:
```

Comment:
Do you mean if someone was developing locally and wanted to upload to a table? My thought was to make it so that running with `Local` would default to not updating the table since most of the time you wouldn't want to do that when developing a new feature. If you need to test the table updating functionality you can use another `Platform` value and your own testing table like I did while working on this.